tasktiger/tasktiger.py

```diff
@@ -89,6 +93,38 @@
 STRING <prefix>:qlock:<queue> (Legacy queue locks that are no longer used)
 """
 
+P = ParamSpec("P")
+R = TypeVar("R")
+
+class TaskCallable(Generic[P, R]):
+    def __init__(self, func: Callable[P, R], tiger: "TaskTiger"):
+        functools.update_wrapper(self, func)
+        self._func = func
+        self._tiger = tiger
+
+        # Default values for attributes
+        self._task_hard_timeout: float | None = None
+        self._task_queue: str | None = None
+        self._task_unique: bool | None = None
+        self._task_unique_key: Collection[str] | None = None
+        self._task_lock: bool | None = None
+        self._task_lock_key: Collection[str] | None = None
+        self._task_retry: int | None = None
+        self._task_retry_on: Collection[type[BaseException]] | None = None
+        self._task_retry_method: Callable[[int], float] | Tuple[Callable[..., float], Tuple] | None = None
+        self._task_batch: bool | None = None
+        self._task_schedule: Callable | None = None
+        self._task_max_queue_size: int | None = None
+        self._task_max_stored_executions: int | None = None
+        self._task_runner_class: type | None = None
+
+    def __call__(self, *args: P.args, **kwargs: P.kwargs) -> R:
+        return self._func(*args, **kwargs)
+
+    def delay(self, *args: P.args, **kwargs: P.kwargs) -> "Task":
+        """Queues a task. See README.rst for an explanation of the options."""
```

Comment:
What _are_ the options? The way I see it, these are just task arguments, right?

Reply:
Looks like this doesn't/didn't support any options.

Reply:
But we pass them as `args=` / `kwargs=` and now `*args, **kwargs` so you wouldn't be able to access those right?